patches/tModLoader/Terraria.ModLoader.Core/ModCompile.cs

```diff
@@ -376,7 +376,7 @@ private class BuildingMod : LocalMod
 
 			// add dll references from the bin folder
 			var libFolder = Path.Combine(mod.path, "lib");
-			foreach (var dllPath in mod.properties.dllReferences.Select(dllName => DllRefPath(mod, dllName)))
+			foreach (var dllPath in mod.properties.dllReferences.SelectMany(dllName => DllRefPathsForPackaging(mod, dllName)))
```

Comment:
The only purpose of this code branch is to copy dll references from the directory provided by the `-eac` flag. This allows people building with vs to ignore the lib folder. Platform dependent dll paths are only in the `lib` folder so they'll just get ignored in the next `if` statement. The entire `lib` folder is packaged wholesale regardless. 

This means you can probably remove `DllRefPathsForPackaging` and combine `DllRefPathForCompilation` with `DllRefPath`